### PR TITLE
Updates to the kubernetes-master README.md

### PR DESCRIPTION
**Pull request by mbruzek**
The kubernetes-master README.md was very out of date. I updated the document with this pull request and plan to move sections of this to the worker, please review and let me know how to write better English words.

### CODE REVIEW

File: cluster/juju/layers/kubernetes-master/README.md [hunk 1]
```diff
@@ -1,118 +1,102 @@
-# kubernetes
+# kubernetes-master
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/README.md]: *Kubernetes-master

File: cluster/juju/layers/kubernetes-master/README.md [hunk 2]
```diff
@@ -1,118 +1,102 @@
...
-The domain name to use for the Kubernetes cluster by the skydns service.
-
-#### enable-dashboard-addons
-
-Enables the installation of Kubernetes dashboard, Heapster, Grafana, and
-InfluxDB.
+# Deployment
 
-# Additional Information
+This charm is not self contained, other charms are needed to model a complete
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/README.md]: Seems like you've included this information twice, but went into more depth here. I'm not sure if you should just combine the two, or leave it as is. I leave that up to your judgement as the author of the intent of these messsages.

File: cluster/juju/layers/kubernetes-master/README.md [hunk 3]
```diff
@@ -1,118 +1,102 @@
...
+This charm was built from other charm layers using the Juju reactive framework.
+The other layers focus on specific subset of operations making this layer 
+specific to operations of Kubernetes master processes.
 
-# Configuration
-For your convenience this charm supports some configuration options to set up
-a Kubernetes cluster that works in your environment:  
+This charm is one part of what is needed to model a Kubernetes cluster. If you
+wish to deploy a complete cluster please look at the collection of charms that
+model a [Kubernetes core](https://jujucharms.com/kubernetes-core/) cluster.
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/README.md]: This should probably be CDK as its our primary point-to-source for kubernetes reference architecture. The core is our developer friendly kick the tires on few machines bundle.

File: cluster/juju/layers/kubernetes-master/README.md [hunk 4]
```diff
@@ -1,118 +1,102 @@
...
+# Resources
 
+The kubernetes-master charm takes advantage of the [Juju Resources](https://jujucharms.com/docs/2.0/developer-resources) 
+feature.
 
-# Storage
-The kubernetes charm is built to handle multiple storage devices if the cloud
-provider works with
-[Juju storage](https://jujucharms.com/docs/devel/charms-storage).
+The resources are already attached to the Juju Charm Store, but in cases where
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/README.md]: We need to be fairly careful about this messaging. We dont give end users any information on how to package their own release of Kubernetes. Perhaps we change the byline to this:

```
Kubernetes releases being delivered in the charms are evaluated by the Canonical ~containers team. They are provided for you by the charms in the Charm Store. If you are deploying juju in a network limited cloud, you will be responsible for creating these resources, or fetching them before deployment.

... insert additional info here if applicable ...

```

I dont know thats any better than whats here, but just a suggestion.

File: cluster/juju/layers/kubernetes-master/README.md [hunk 5]
```diff
@@ -1,118 +1,102 @@
...
 
-**kubectl.package.created** - Indicates the availability of the `kubectl`
-application along with the configuration needed to contact the cluster
-securely. You will need to download the `/home/ubuntu/kubectl_package.tar.gz`
-from the kubernetes leader unit to your machine so you can control the cluster.
+# Adding optional storage
 
-**skydns.available** - Indicates when the Domain Name System (DNS) for the
-cluster is operational.
+The kubernetes-master charm handles different persistent storage devices that
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/README.md]: I feel like this paragraph should be nixed, and we should keep the link in the following short paragraph, where we redirect users to the CDK bundle to evaluate deploying ceph based storage, and we'll keep that readme up to date as more storage backends are added.

but we're not doing anything with traditional juju storage feature at this time... so this seems mostly irrelevant.

File: cluster/juju/layers/kubernetes-master/README.md [hunk 6]
```diff
@@ -1,118 +1,102 @@
...
  - [Kubernetes issue tracker](https://github.com/kubernetes/kubernetes/issues)
- - [Kubernetes Documenation](http://kubernetes.io/docs/)
+ - [Kubernetes documentation](http://kubernetes.io/docs/)
  - [Kubernetes releases](https://github.com/kubernetes/kubernetes/releases)
 
 # Contact
 
- * Charm Author: Matthew Bruzek &lt;Matthew.Bruzek@canonical.com&gt;
- * Charm Contributor: Charles Butler &lt;Charles.Butler@canonical.com&gt;
+The kubernetes-master charm is free and open source operations software created
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/README.md]: I like this footer. +1

File: cluster/juju/layers/kubernetes-master/README.md [hunk 7]
```diff
@@ -1,118 +1,102 @@
...
+by the containers team at Canonical. 
+
+Canonical also offers enterprise support and customisation services. Please
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/README.md]: http://www.dictionary.com/browse/customization

File: cluster/juju/layers/kubernetes-master/README.md (hunk 1 above)
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-master/README.md]: Errors on the first line!

File: cluster/juju/layers/kubernetes-master/README.md (hunk 2 above)
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-master/README.md]: I see the duplication now that you pointed it out. Not intentional, will re-factor.

File: cluster/juju/layers/kubernetes-master/README.md (hunk 5 above)
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-master/README.md]: The kubernetes-master is the only charm with the storage action. I feel if they are seeing errors with storage they may come to this readme for more information. The CDK bundle README does contain the instructions on how to deploy Ceph (which I agree is outside the scope of this charm's README). 

I just wanted to introduce the feature and give them the command to run because it is unique to this charm.

File: cluster/juju/layers/kubernetes-master/README.md (hunk 7 above)
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-master/README.md]: I copied the en-uk way of spelling customisation from the page that was linked. Not sure what to do here, our README should agree with the marketing page.

**Comment by chuckbutler** [comment]
Related to:  https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/99

**Comment by Cynerva** [comment]
I don't have anything to add that @chuckbutler hasn't covered already. Thanks for taking the time to do this @mbruzek, good updates :+1:

File: cluster/juju/layers/kubernetes-master/README.md [hunk 8]
```diff
@@ -1,118 +1,96 @@
...
-a Kubernetes cluster that works in your environment:  
+This charm was built from other charm layers using the Juju reactive framework.
+The other layers focus on specific subset of operations making this layer 
+specific to operations of Kubernetes master processes.
 
-#### dns_domain
+# Deployment
 
-The domain name to use for the Kubernetes cluster by the skydns service.
+This charm is not fully function when deployed by itself. It requires other
```
Comment (Cynerva) [inline cluster/juju/layers/kubernetes-master/README.md]: functional*

Other than that nitpick, all looks good to me!

Comment (mbruzek) [inline cluster/juju/layers/kubernetes-master/README.md]: good catch

**Comment by mbruzek** [comment]
I have updated the comments from both reviewers please re-review when you have time. Thanks.

**Comment by chuckbutler** [comment]
+1 LGTM
